modules/backend/behaviors/RelationController.php

```diff
@@ -1472,33 +1471,75 @@ protected function evalToolbarButtons()
     {
         $buttons = $this->getConfig('view[toolbarButtons]');
 
-        if ($buttons === false) {
-            return null;
-        }
-        elseif (is_string($buttons)) {
-            return array_map('trim', explode('|', $buttons));
-        }
-        elseif (is_array($buttons)) {
-            return $buttons;
+        if (!is_array($buttons)) {
+            if ($buttons === false) {
+                return null;
+            } elseif (is_string($buttons)) {
+                $buttons = array_map('trim', explode('|', $buttons));
+            } elseif (is_array($buttons)) {
+                $buttons = $buttons;
```

Comment:
@DanHarrin you might want to drop this bit of code - it is unreachable with the surrounding condition.
```suggestion
```